examples/offline/fetch_her_ddpg.py

```diff
@@ -0,0 +1,265 @@
+from tianshou.exploration import GaussianNoise
+from tianshou.policy import DDPGPolicy
+from tianshou.trainer import offpolicy_trainer
+from tianshou.utils import TensorboardLogger, WandbLogger
+from tianshou.utils.net.common import Net
+from tianshou.utils.net.continuous import Actor, Critic
+
+
+def get_args():
+    # python3 fetch_her_ddpg.py --task FetchReach-v3 --seed 0 --horizon 50
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--task", type=str, default="FetchReach-v3")
+    parser.add_argument("--seed", type=int, default=0)
+    parser.add_argument("--buffer-size", type=int, default=1000000)
+    parser.add_argument("--hidden-sizes", type=int, nargs="*", default=[256, 256])
+    parser.add_argument("--actor-lr", type=float, default=1e-3)
+    parser.add_argument("--critic-lr", type=float, default=1e-3)
+    parser.add_argument("--gamma", type=float, default=0.99)
+    parser.add_argument("--tau", type=float, default=0.005)
+    parser.add_argument("--exploration-noise", type=float, default=0.1)
+    parser.add_argument("--start-timesteps", type=int, default=25000)
+    parser.add_argument("--epoch", type=int, default=200)
+    parser.add_argument("--step-per-epoch", type=int, default=5000)
+    parser.add_argument("--step-per-collect", type=int, default=1)
+    parser.add_argument("--update-per-step", type=int, default=1)
+    parser.add_argument("--n-step", type=int, default=1)
+    parser.add_argument("--batch-size", type=int, default=256)
+    parser.add_argument("--horizon", type=int, default=50)
+    parser.add_argument("--future-k", type=int, default=8)
+    parser.add_argument("--training-num", type=int, default=1)
+    parser.add_argument("--test-num", type=int, default=10)
+    parser.add_argument("--logdir", type=str, default="log")
+    parser.add_argument("--render", type=float, default=0.)
+    parser.add_argument(
+        "--device", type=str, default="cuda" if torch.cuda.is_available() else "cpu"
+    )
+    parser.add_argument("--resume-path", type=str, default=None)
+    parser.add_argument("--resume-id", type=str, default=None)
+    parser.add_argument(
+        "--logger",
+        type=str,
+        default="tensorboard",
+        choices=["tensorboard", "wandb"],
+    )
+    parser.add_argument("--wandb-project", type=str, default="HER-benchmark")
+    parser.add_argument(
+        "--watch",
+        default=False,
+        action="store_true",
+        help="watch the play of pre-trained policy only",
+    )
+    return parser.parse_args()
+
+
+def make_fetch_env(task, training_num, test_num):
+    env = TruncatedAsTerminated(gym.make(task))
+    train_envs = ShmemVectorEnv(
+        [lambda: TruncatedAsTerminated(gym.make(task)) for _ in range(training_num)]
+    )
+    test_envs = ShmemVectorEnv(
+        [lambda: TruncatedAsTerminated(gym.make(task)) for _ in range(test_num)]
+    )
+    return env, train_envs, test_envs
+
+
+class DictStateNet(Net):
```

Comment:
I'm a bit concerned about the fact that the user need to follow the example in this file closely in order to apply HER to any other task. Can you rewrite `DictStateNet` as a [decorator](https://en.wikipedia.org/wiki/Decorator_pattern) of `Net` so that it can be applied to any `Net`? The way I see it, this decorator should reside in [common.py](https://github.com/thu-ml/tianshou/blob/master/tianshou/utils/net/common.py) or [continous.py](https://github.com/thu-ml/tianshou/blob/master/tianshou/utils/net/continuous.py) as a utility function to wrap any `Net` which wants to work with `HERReplayBuffer`.